src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2354,6 +2352,150 @@ private uint getClassNumInstanceFields(CORINFO_CLASS_STRUCT_* cls)
             throw new InvalidOperationException();
         }
 
+        private GetTypeLayoutResult GetTypeLayoutHelper(MetadataType type, uint parentIndex, uint baseOffs, FieldDesc field, CORINFO_TYPE_LAYOUT_NODE* treeNodes, nuint maxTreeNodes, nuint* numTreeNodes)
+        {
+            if (*numTreeNodes >= maxTreeNodes)
+            {
+                return GetTypeLayoutResult.Partial;
+            }
+
+            uint structNodeIndex = (uint)(*numTreeNodes)++;
+            CORINFO_TYPE_LAYOUT_NODE* parNode = &treeNodes[structNodeIndex];
+            parNode->simdTypeHnd = null;
+            parNode->diagFieldHnd = field == null ? null : ObjectToHandle(field);
+            parNode->parent = parentIndex;
+            parNode->offset = baseOffs;
+            parNode->size = (uint)type.GetElementSize().AsInt;
+            parNode->numFields = 0;
+            parNode->type = CorInfoType.CORINFO_TYPE_VALUECLASS;
+            parNode->hasSignificantPadding = false;
+
+#if READYTORUN
+            // The contract of getTypeLayout is carefully crafted to still
+            // allow us to return hints about fields even for types outside the
+            // version bubble. The general idea is that the JIT does not use
+            // the information returned by this function to create new
+            // optimizations out of the blue, but only as a hint to optimize
+            // existing field uses more thoroughly. In particular the uses of
+            // fields outside the version bubble are only non-opaque and
+            // amenable to the optimizations that this unlocks if they already
+            // went through EncodeFieldBaseOffset.
+            //
+            if (!_compilation.IsLayoutFixedInCurrentVersionBubble(type))
+            {
+                // For types without fixed layout the JIT is not allowed to
+                // rely on padding bits being insignificant, since fields could
+                // be added later inside that padding without invalidating the
+                // generated code.
+                parNode->hasSignificantPadding = true;
+            }
+#endif
+
+            if (type.IsExplicitLayout || (type.IsSequentialLayout && type.GetClassLayout().Size != 0) || type.IsInlineArray)
+            {
+                if (!type.ContainsGCPointers && !type.IsByRefLike)
+                {
+                    parNode->hasSignificantPadding = true;
+                }
+            }
+
+            // The intrinsic SIMD/HW SIMD types have a lot of fields that the JIT does
+            // not care about since they are considered primitives by the JIT.
+            if (type.IsIntrinsic)
+            {
+                string ns = type.Namespace;
+                if (ns == "System.Runtime.Intrinsics" || ns == "System.Numerics")
+                {
+                    parNode->simdTypeHnd = ObjectToHandle(type);
+                    if (parentIndex != uint.MaxValue)
```

Comment:
We should only set this if NeedsTypeLayoutCheck returns false for the type.

Reply:
If that returns true we should probably just fail getting this entire type layout.

You will need to put that under a #if READYTORUN

Reply:
Addressed.